### ISSUE TITLE
Character set of PID tstr fields such as family_name should be explicitly defined as Unicode

### ISSUE THREAD

**Issue by joelposti** [eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework#157]
Character set of the PID's tstr attributes, such as family_name and given_name, is currently undefined in PID Rule Book 1.0.0. It should be explicitly defined. Considering that so much of the PID Rule Book is inspired by ISO/IEC
18013-5 Mobile driving licence specification there is a real risk that PID providers and other parties only implement support for latin1 character set as that is defined in ISO/IEC 18013-5 as the character set for family_name and given_name attributes. I and in extension DVV (the Finnish Digital and Population Data Services Agency) that I represent, object to latin1 being the character set for PID's tstr attributes, because latin1 is too constrained to allow correct representation of Sámi names (among others). 

We propose that the character set of the PID's tstr attributes should be defined explicitly in the PID Rule Book and that the character set should be Unicode in its full range. It would be even better if the character set was explicitly defined in the ARF for all EUDIW-compliant attestations.

It is important to understand the difference between a character set and a character encoding. A character set is a set of characters that are available to use. A character encoding, on the other hand, is the way those characters are represented in memory as binary. Character **encoding** of the PID's tstr attributes is implicitly UTF-8, because both [the JSON specification](https://www.rfc-editor.org/rfc/rfc8259#section-8.1) and [the CBOR specification](https://www.rfc-editor.org/rfc/rfc8949.html#section-3.1) define UTF-8 as the character encoding. PID Rule Book also defines UTF-8 as the character **encoding** for the tstr attributes in the SD-JWT PID. However, the character **set** is not defined which we fear results in latin1 being inherited from ISO/IEC 18013-5.

**Comment by digeorgi** [comment]
Thank you very much for your input.

For mdoc format, the character set of the PID's tstr attributes is inherited from ISO/IEC 18013-5.
For other format, where the character set is not inherited from existing standards (e.g., SD-JWT VC), the character set of the PID's tstr attributes will be defined explicitly in the PID Rule Book as Unicode in its full range.

**Comment by joelposti** [comment]
> Thank you very much for your input.
> 
> For mdoc format, the character set of the PID's tstr attributes is inherited from ISO/IEC 18013-5. For other format, where the character set is not inherited from existing standards (e.g., SD-JWT VC), the character set of the PID's tstr attributes will be defined explicitly in the PID Rule Book as Unicode in its full range.

Thank you for your response!

We are happy that the character set of SD-JWT PID tstr attributes will be Unicode in its full range.

However, we are not so happy with the notion that the character set of mdoc PID tstr attributes will be inherited from ISO/IEC 18013-5. Could you provide argumentation as to why you would like to constrain mdoc PID tstr attributes to latin1? What is being defined here is new technology and a completely new type of credential (PID). Why do we need to consider legacy character sets such as latin1?

**Comment by joelposti** [comment]
Hello @digeorgi! Would you have the time and opportunity to check the questions I posed in my previous comment? I am still worried about mdoc PID tstr attributes having latin1 as their character set. I would like them to have Unicode as their character set.

I noticed you commented some time ago in another issue: https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/issues/220#issuecomment-2469782733. It seems your comment there contradicts what you commented earlier in this issue (mdoc PID tstr attributes inheriting latin1 character set from ISO/IEC 18013-5). Or maybe I am not understanding something.

> As said above, tstr attributes support the full Unicode character set. Issuing countries can therefore use their national characters for names. It is up to the Relying Party to transcribe this to Latin1 or any other characters, if they so choose.

Could you clarify what is the character set of the mdoc PID tstr attributes?

**Comment by david-bakker** [comment]
Hi Joel, when doing some cleaning up I discovered that we never came back to this issue. My apologies!

Anyway, you are right, our initial answer was wrong. The intention has always been to support the full Unicode range for both ISO-compliant and SD-JWT VC-compliant PIDs. In fact, that is why we chose to not include in the PID Rulebook the given_name_national_character and family_name_national_character attributes defined in ISO/IEC 18013-5. Section 4.2.1 of the PID Rulebook in ARF 1.5.0 specifies that all attributes having encoding format tstr SHALL be encoded in UTF-8 and SHALL support the full Unicode range.

**Comment by joelposti** [comment]
Hello David! Fantastic, thank you for the clarification and for pointing out that since ARF 1.5.0 Unicode has been explicitly defined as the character set for mdoc PID tstr attributes. I quote it below for documentative reasons:

> https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/blob/v1.5.0/docs/annexes/annex-3/annex-3.01-pid-rulebook.md#42-encoding-of-pid-attributes-and-metadata
>
> Regarding type tstr: this document confirms that, as specified in RFC 8949, a tstr SHALL be encoded in UTF-8 and SHALL support the full Unicode range.

The remaining request from me and DVV is to have a similar explicit character set requirement for the SD-JWT VC-based/JSON-based PID in section ‘5 SD-JWT VC-based encoding of PID – 5.2 Encoding of PID attributes’ when content related Topic V – PID Rulebook are added to the ARF and the PID Rulebook.

**Comment by david-bakker** [comment]
Yes, it is definitely the intent to require UTF-8 and the full Unicode set for JSON strings as well.

**Comment by joelposti** [comment]
Hello @david-bakker! I noticed that ARF 1.9.0 PID Rulebook now contains the following sentence concerning SD-JWT VC-complaint PID:

> https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/blob/v1.9.0/docs/annexes/annex-3/annex-3.01-pid-rulebook.md#41-encoding-of-pid-attributes-and-metadata
>
> A JSON string used in an SD-JWT VC-encoded PID SHALL be encoded in UTF-8 and SHALL support the full Unicode range, unless explicitly specified otherwise in the tables below or the references therein.

This is in addition with the following sentence concerning ISO/IEC 18013-5-complaint PID (already added in ARF 1.5.0):

> https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/blob/v1.9.0/docs/annexes/annex-3/annex-3.01-pid-rulebook.md#31-encoding-of-pid-attributes-and-metadata
>
> Regarding type tstr: this document confirms that, as specified in RFC 8949, a tstr SHALL be encoded in UTF-8 and SHALL support the full Unicode range.

I and DVV are happy with these explicit character set definitions. Thank you! This issue can be closed as completed.
